substrate/frame/system/src/lib.rs

```diff
@@ -320,121 +320,6 @@ pub mod pallet {
 			type PostInherents = ();
 			type PostTransactions = ();
 		}
-
-		/// Default configurations of this pallet in a solochain environment.
-		///
-		/// ## Considerations:
-		///
-		/// By default, this type makes the following choices:
-		///
-		/// * Use a normal 32 byte account id, with a [`DefaultConfig::Lookup`] that implies no
-		///   'account-indexing' pallet is being used.
-		/// * Given that we don't know anything about the existence of a currency system in scope,
-		///   an [`DefaultConfig::AccountData`] is chosen that has no addition data. Overwrite this
-		///   if you use `pallet-balances` or similar.
-		/// * Make sure to overwrite [`DefaultConfig::Version`].
-		/// * 2s block time, and a default 5mb block size is used.
-		pub struct SolochainDefaultConfig;
-
-		#[frame_support::register_default_impl(SolochainDefaultConfig)]
-		impl DefaultConfig for SolochainDefaultConfig {
```

Comment:
what is the downside if we keep `ParachainDefaultConfig`/`SolochainDefaultConfig` but it  keeps only the uncontroversial stuff, like `PalletInfo`, `Runtime*`? 

Seems a bit backwards to me, although I have not followed the conversation where this was concluded.

Reply:
What I see here is also that there were indeed mistakes in the previous version of this type: 

1. `type AccountId` should be `Multi*` for maximum compatibility
2. `type SystemWeightInfo/WeightInfo = (); ` is not a good production advice. 
3. same as `type SS58Prefix` and `Version`, you don't want a production runtime to have these set to `()`

Reply:
> what is the downside if we keep `ParachainDefaultConfig`/`SolochainDefaultConfig` but it keeps only the uncontroversial stuff, like `PalletInfo`, `Runtime*`?

The major concern here was the impact on downstream impls as the defaults could be added/modified without their knowledge. The current infra for `derive_impl` needs us to provide defaults for all params that are not marked with `#[pallet::no_default]` and its not possible to choose these for each config. So, `ParachainDefaultConfig` would still need to override all such params, which might eventually find there way in the downstream projects.

It would be possible if we could annotate `#[pallet::no_default]` with a config type. This would allow us to keep most params open to defaults for TestConfigs but restricted for others. 

> I think there is still a cateogry of people that want to build a real runtime, perhaps for a Local network, so they want realistic, but non-production types. TestDefaultConfig will not be okay for this.

As long as it is for a local network, why would `TestDefaultConfig` not be okay? I believe it clearly signifies the fact that it should be changed before going to production?

Reply:
Could a good compromise be to just have a `TestDefaultConfig` in the pallet itself, and then have a config shared by our testnet system parachains in `testnets-common` or `parachains-common` (something that I started to do in #4056 anyway) which people could then use and inherit from as a sensible baseline. Likewise for the testnet relays. The fellowship repo would have its own `DefaultConfig`s defined for production networks.
For extra visibility of changes we could even version the config so people would be aware of changes to existing values, but when we add a new config item we could just use a new sensible value within the same version.
WDYT? This changes the contract from "here's a stable sensible baseline for you to adapt which we'll try to keep up to date" to "this is the config we use for the actual system chains"

Reply:
> then have a config shared by our testnet system parachains in testnets-common or parachains-common (something that I started to do in https://github.com/paritytech/polkadot-sdk/pull/4056 anyway) which people could then use and inherit from as a sensible baseline.

Yes, this makes sense to me. I think @kianenigma also previously mentioned keeping the configs separate that could then be reused by other folks as needed.

Reply:
> I see. This is a real issue, but I am inclined to suggest that there should be a trick to just fill these things that have to be overwritten in with a `struct UnImplemented`, which would do nothing other than raising a compiler error, prompting users to overwrite it.

Let me think about this a bit more.

> `TestDefaultConfig` is optimized for in-code testing, and sometimes this diverges from real-life-testing. For example the `AccountId` type is one instance of this.

True, but I was imagining this to be a minimal change. For example, I had to override `AccountId` and `Lookup` to make it work with the minimal template.

Reply:
@seadanda I like your idea, should be useful for system chains